specs/0106-concordance-search.md

```diff
@@ -0,0 +1,27 @@
+- Feature Name: Concordance Search
+- Created: 2020-07-07
+- Associated Bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1514698
+
+# Summary
+
+Add ability to search existing translations while translating or reviewing strings.
+
+# Motivation
+
+Having the ability to search for existing translations is a crucial tool for assuring translation consistency. It allows you to see how was the same expression translated in the past or verify that the translation you intend to use is consistent with the rest of the corpus. Pontoon tries to automate that as much as possible with Machinery suggestions by querying Translation Memory as soon as a string is opened for translation.
+
+That being said, Translation Memory is only queried for strings as a whole, even if you "Search in Machinery". If you're interested in the translation of a word "file", but your corpus only uses that word as part of longer strings like "Select a file..." or "Source file uploaded", Machinery won't find any matches.
+
+Alternativelly, you can switch to the All Projects view and use the string list search, which searches for substrings within existing translations for each word separatelly, but that also comes with a few drawbacks. You either lose translation progress in the editor and the state of the string list, or you need to use a separate tab. You also won't be able to search for translations in the projects that aren't active anymore.
+
+# Feature explanation
+
+This feature changes the way "Search in Machinery" searches Translation Memory. Instead of searching for fuzzy matches of whole strings stored in Translation Memory, `source` and `target` fields of the `TranslationMemoryEntry` model are searched for each keyword and each phrase within quotes separately.
```

Comment:
I'm completely lost on this explanation: can you provide an example? To quote the example above, would you need to search for `"file"`? Would `"select" "file"` work?

Reply:
Check out the next paragraph. In essence, search will work the same as in the string list, except that we'll only search `source` and `target` fields. I'll try to come up with a clearer explanation.